### PR TITLE
fix(metrics): Add TASKS_ADDED_COUNT and TASKS_RESET_COUNT metrics in router scheduler flow

### LINKED ISSUE #2090: [BUG] Add metrics in router scheduler flow as it is being ignored now
### Bug Description

Router is not maintaining any metrics about the task added in scheduler. this has to be fixed
File: crates/router/src/core/payments/helpers.rs
Method: add_domain_task_to_pt

### Expected Behavior

Metrics has to be added to know the TASKS_ADDED_COUNT and TASKS_RESET_COUNT

### Actual Behavior

Currently those are not available

### Steps To Reproduce

NA

### Context For The Bug

_No response_

### Environment

NA

### Have you spent some time checking if this bug has been raised before?

- [X] I checked and didn't find a similar issue

### Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md)

### Are you willing to submit a PR?

None

### PR DESCRIPTION
**Pull request by pixincreate**
## Type of Change

- [x] Bugfix

## Description

Currently `TASKS_ADDED_COUNT` and `TASKS_RESET_COUNT` metrics does not exist in router to track the tasks added in scheduler. This PR aims at adding that.  
Closes #2090

## Motivation and Context

We need to track the tasks added in scheduler.

## How did you test it?

Did a few payments (Ran Stripe and Volt collection a few times):
image

### Local setup:

Modify 3 files:
- `docker-compose.yml`: Expose the `otel-collector` port by replacing `"4317"` with `"4317:4317"`
- `development.toml`: Enable logging by setting `metrics_enabled` to `true` and `otel_exporter_otlp_endpoint` to `http://localhost:4317`
- `otel-collector.yml`: Add logging in exporters of pipeline service: `[prometheus, logging]` instead of `[prometheus]`
- Open a terminal window, start server by executing: `cargo r`
- Create another terminal window for psql, execute:
	```
	psql hyperswitch_db
	\x
	\dt
	table process_tracker;
	``` 
- Create another 3 terminal windows. One for consumer, another for producer and another for Grafana. Execute:
	```
	# In consumer terminal window
	SCHEDULER_FLOW=consumer cargo r --bin scheduler 
	# In producer terminal window
	SCHEDULER_FLOW=producer cargo r --bin scheduler
	# In Grafana terminal window
	docker compose up -d grafana prometheus otel-collector
	```
- Execute `curl telnet://localhost:4317` in yet another terminal window to see if  this URL can be accessed or not
- Grafana username/password - admin/admin 
  > incase reset is required, execute in terminal:
  > ` grafana-cli admin reset-admin-password <new-password>`

In env:
- Open Grafana > VM Single > `Metric:router_TASKS_ADDED_COUNT_total`, `label:job=router` > `Run Query`

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/payments/helpers.rs [hunk 1]
```diff
@@ -987,14 +987,14 @@ where
         match schedule_time {
             Some(stime) => {
                 if !requeue {
-                    // scheduler_metrics::TASKS_ADDED_COUNT.add(&metrics::CONTEXT, 1, &[]); // Metrics
+                    metrics::TASKS_ADDED_COUNT.add(&metrics::CONTEXT, 1, &[]); // Metrics
```
Comment (SanchithHegde) [inline crates/router/src/core/payments/helpers.rs:990]: Nit: we can remove the comment here and on the other line

**Review by SanchithHegde — DISMISSED** [review]

File: crates/router/src/core/payments/helpers.rs [hunk 2]
```diff
@@ -987,14 +987,16 @@ where
         match schedule_time {
             Some(stime) => {
                 if !requeue {
-                    // scheduler_metrics::TASKS_ADDED_COUNT.add(&metrics::CONTEXT, 1, &[]); // Metrics
+                    // Increment the count of added tasks every time a payment has been confirmed or PSync has been called
```
Comment (SanchithHegde) [inline crates/router/src/core/payments/helpers.rs:990]: Task addition is not limited to a payment being confirmed or a payment sync occurring though.

File: crates/router/src/core/api_keys.rs [hunk 3]
```diff
@@ -270,6 +270,14 @@ pub async fn add_api_key_expiry_task(
                 api_key_expiry_tracker.key_id
             )
         })?;
+    metrics::TASKS_ADDED_COUNT.add(
+        &metrics::CONTEXT,
+        1,
+        &[metrics::request::add_attributes(
+            "api_key_expiry",
```
Comment (SanchithHegde) [inline crates/router/src/core/api_keys.rs:277]: Please keep the key the same everywhere `TASKS_ADDED_COUNT` and `TASKS_RESET_COUNT` are being incremented. That way filtering becomes easier:

If you'd use `flow` for the key, the attributes would be `flow = api_key_expiry`, `flow = delete_tokenize_data`, `flow = payments_confirm`, etc. as compared to `api_key_expiry = 7` and `delete_tokenize_data = DELETE_TOKENIZE_DATA`, which might not make much sense.

**Review by SanchithHegde — APPROVED** [review]

File: crates/router/src/core/payment_methods/vault.rs [hunk 4]
```diff
@@ -984,6 +984,14 @@ pub async fn start_tokenize_data_workflow(
...
                 .finish_with_status(db.as_scheduler(), format!("COMPLETED_BY_PT_{id}"))
                 .await?;
+            metrics::TASKS_ADDED_COUNT.add(
+                &metrics::CONTEXT,
+                1,
+                &[metrics::request::add_attributes(
+                    "flow",
+                    "DeleteTokenizeData",
+                )],
+            );
```
Comment (Narayanbhat166) [inline crates/router/src/core/payment_methods/vault.rs:994]: This should be when the task is being inserted into the process tracker

**Review by Narayanbhat166 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by sai-harsha-vardhan — APPROVED** [review]

**Pull request merged**